pingpong/lib/pingpong.ex

```diff
@@ -0,0 +1,16 @@
+defmodule Pingpong do
+  @moduledoc """
+  Example of a simple pingpong server
+  """
+
+  @doc """
+  Runs pingpong server
+  """
+  def server do
+    receive do
```

Comment:
❤

You would barely need the low-level `receive/1` in real life until you need it, and when you need, you are to find it amazing. Yes, we have abstractions like `GenServer` on top of it, but there are still use-cases when this clause does its job really great.

Any asynchronous call might be turned into synchronous in-place, without a huge refactoring, but simply passing a pid, sending a message back upon completion, and awaiting for this message. Like

```elixir
# BEFORE
@impl GenServer
def handle_cast(:heavy_work, state) do
  state = do_heavy_work(state)
  {:noreply, state}
end
# called as
GenServer.cast(pid, :heavy_work)
```
need to make it synchronous?—Nothing is easier
```elixir
# AFTER
@impl GenServer
def handle_cast({:heavy_work, pid}, state) do
  handle_cast(:heavy_work, pid)
  |> tap(fn {_, new_state} -> send(pid, {:heavy_work, new_state}) end)
end
# no changes to the original function
def handle_cast(:heavy_work, state) do
  state = do_heavy_work(state)
  {:noreply, state}
end
# called as
GenServer.cast(pid, {:heavy_work, self()})
receive do
  {:heavy_work, result} -> do_handle_result(result)
end
```